assets/src/blocks/Articles/ArticlesFrontend.js

```diff
@@ -0,0 +1,134 @@
+import { Component, Fragment } from '@wordpress/element';
+import { ArticlePreview } from './ArticlePreview';
+
+const { apiFetch } = wp;
+const { addQueryArgs } = wp.url;
+
+export class ArticlesFrontend extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      posts: [],
+      total_pages: 0,
+      dataset: {},
+      page: 0
+    };
+
+    this.loadArticles = this.loadArticles.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadArticles();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { article_count, tags, posts, post_types, ignore_categories } = this.props;
+    if (
+      article_count !== prevProps.article_count ||
+      tags.length !== prevProps.tags.length ||
+      posts.length !== prevProps.posts.length ||
+      post_types.length !== prevProps.post_types.length ||
+      ignore_categories !== prevProps.ignore_categories
+    ) {
+      this.loadArticles();
+    }
+  }
+
+  loadArticles(page) {
+    const { article_count, posts, tags, post_types, ignore_categories, postType, postId, setTotalPages } = this.props;
+
+    const args = {
+      article_count,
+      post_types,
+      posts,
+      tags,
+      ignore_categories
+    };
+
+    if (page) args.offset = page * article_count;
+
+    if (postType === 'post') args.exclude_post_id = postId;
+
+    const queryArgs = {
+      path: addQueryArgs('/planet4/v1/get-articles', args)
+    };
+    apiFetch(queryArgs).then(({ recent_posts, total_pages }) => {
+      if (page) {
+        this.setState({
+          posts: [...this.state.posts, ...recent_posts],
+          page
+        })
+      } else {
+        this.setState({
+          posts: recent_posts,
+          total_pages
+        });
+        if (setTotalPages) setTotalPages(total_pages);
+      }
+    });
+  }
+
+  render() {
+    const {
+      article_heading,
```

Comment:
This is a bit complicated to change, because the frontend is rendered via [this file](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/assets/src/frontendIndex.js) and changing the behavior there would mean changing it for other WYSIWYG blocks too...

Reply:
Maybe we can instead ensure that the frontend block only gets the attributes? As that will be the only data it should have available in the frontend. That way we also keep editor and frontend code separated. If the editor needs to render something else than the frontend, it's probably better to split the frontend node in multiple parts. Then we can just include only the parts we need in the editor, and we don't need the conditionals in our frontend component.

Reply:
That would be ideal 😄 I've already removed the `isEditing`, but there is actually an issue for `postType` since it's passed in the editor via `withSelect`... It's not present in the frontend 😱 I guess we should add it to the attributes...?